### ISSUE TITLE
Inconsistencies between evaluation after training and loading the model from checkpoint

### ISSUE THREAD

**Issue by Kromid** [ThilinaRajapakse/simpletransformers#1023]
Hi everyone,

I think that I found an inconsistency in the evaluation method. Please see the following MWE

```

from simpletransformers.classification import ClassificationModel
from sklearn.metrics import  accuracy_score

# Load your data here: train_df and test_df

# Define model

train_args ={"num_train_epochs": 10}
model = ClassificationModel("bert", "bert-base-german-cased", num_labels=2,args=train_args,use_cuda=True)
model.train_model(train_df = train_df)

# evaluate trained model
result, model_outputs, wrong_predictions = model.eval_model(test_df, acc=accuracy_score)
print(result)

# Load model from checkpoint
model = ClassificationModel("bert", "outputs/... load model after checkpoint 10", num_labels=2,args=train_args,use_cuda=True)
result, model_outputs, wrong_predictions = model.eval_model(test_df, acc=accuracy_score)
print(result)

```

The evaluation of the model directly after training gives a completely different result than loading the model from the ckeckpoint (In my case 65% acc vs. 90% acc.)

**Comment by AbinayaM02** [comment]
@Kromid : eval_model uses the model that's completely trained through all the epochs (the model is from the final checkpoint). So, the accuracy here might be different from the accuracy of the model loaded at a particular checkpoint.

**Comment by Kromid** [comment]
But if I train 10 epochs and then I load the model after this checkpoint (Named: "checkpoint-XYZ-epoch-10" shouldn't it be the same? This is what I meant above.
I am worried since the accuracy difference is so huge.
